rules/integrations/azure/defense_evasion_entra_suspicious_auth_broker_activity_on_behalf_of_principal_user.toml

```diff
@@ -0,0 +1,153 @@
+- Any additional sign-ins from the `ips` involved, even outside the broker, to determine if tokens have been reused elsewhere.
+
+### False Positive Analysis
+
+- This pattern may occur if the user is registering a new device legitimately from two networks (e.g., mobile hotspot and home).
+- Security software (e.g., endpoint detection tools) or identity clients may produce rapid Graph and DRS access in rare edge cases.
+- Developers or IT administrators working across environments may also produce similar behavior.
+
+### Response and Remediation
+
+- If confirmed unauthorized, revoke all refresh tokens for the affected user and remove any devices registered during this session.
+- Notify the user and determine whether the device join or authentication activity was expected.
+- Audit Conditional Access and broker permissions (`29d9ed98-a469-4536-ade2-f981bc1d605e`) to ensure policies enforce strict access controls.
+- Consider blocking token-based reauthentication to Microsoft Graph and DRS from suspicious locations or user agents.
+- Continue monitoring for follow-on activity like lateral movement or privilege escalation.
+"""
+references = [
+    "https://www.volexity.com/blog/2025/04/22/phishing-for-codes-russian-threat-actors-target-microsoft-365-oauth-workflows/",
+    "https://github.com/dirkjanm/ROADtools",
+    "https://dirkjanm.io/phishing-for-microsoft-entra-primary-refresh-tokens/",
+]
+risk_score = 73
+rule_id = "375132c6-25d5-11f0-8745-f661ea17fbcd"
+setup = """#### Required Microsoft Entra ID Sign-In Logs
+This rule requires the Microsoft Entra ID Sign-In Logs integration be enabled and configured to collect sign-in logs. In Entra ID, sign-in logs must be enabled and streaming to the Event Hub used for the Azure integration.
+"""
+severity = "high"
+tags = [
+    "Domain: Cloud",
+    "Data Source: Azure",
+    "Data Source: Entra ID",
+    "Data Source: Entra ID Sign-in Logs",
+    "Use Case: Identity and Access Audit",
+    "Use Case: Threat Detection",
+    "Resources: Investigation Guide",
+    "Tactic: Defense Evasion",
+    "Tactic: Persistence",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+FROM logs-azure.signinlogs* metadata _id, _version, _index
+
+// filter for Microsoft Entra ID Sign-in Logs
+| where event.dataset == "azure.signinlogs"
+
+    // filters on member principals, excluding service principals
+    and azure.signinlogs.properties.user_type == "Member"
+    and source.ip is not null
+    and azure.signinlogs.identity is not null
+    and azure.signinlogs.properties.user_principal_name is not null
+    and event.outcome == "success"
+
+    // filter for successful sign-ins to Microsoft Graph and DRS/AAD from the Microsoft Authentication Broker
+    and (azure.signinlogs.properties.app_display_name == "Microsoft Authentication Broker" or azure.signinlogs.properties.app_id == "29d9ed98-a469-4536-ade2-f981bc1d605e")
```

Comment:
we may want to add here "aebc6443-996d-45c2-90f0-388ff96faa56" (Visual Studio Code as well)

Reply:
Putting in VSCode first party client would expand the scope of this so it wouldn't be just suspicious auth broker activity on behalf of a user principal. Maybe we have a separate rule where we do some similar aggregations but have a list of common first-party clients that are used in these phishing campaigns? (Teams, Broker, VSCode, Azure Dev, Sharepoint, etc.). Thoughts?

Reply:
++